src/lib.rs

```diff
@@ -44,14 +54,88 @@ impl Reading {
         humidity: String,
         pressure: String,
         dust_concentration: String,
-        air_purity: String
+        air_purity: String,
     ) -> Reading {
         Reading {
             temperature,
             humidity,
             pressure,
             dust_concentration,
-            air_purity
+            air_purity,
+        }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+struct ResponseLog<'a, W: Write> {
```

Comment:
This is great Caleb. Do you have a specific reason to not name this more broadly and only use it for logging responses? Could we use this as our logger more widely? Maybe you could share your thoughts on naming it `Response` specifically?